Idrall/Check.idr

```diff
@@ -376,6 +415,17 @@ mutual
        b' <- evalClosure bT (VNeutral aT (NVar x))
        b <- readBackTyped (extendCtx ctx x aT) (VConst CType) b'
        Right (EPi x a b)
+  readBackTyped ctx (VConst CType) (VHPi i ty f) =
+    let i' = freshen (ctxNames ctx) i in
+    do a <- readBackTyped ctx (VConst CType) ty
+       body' <- readBackTyped ctx (VConst CType) (f (VNeutral ty (NVar i')))
+       -- TODO not remotely sure about this ^^^, especially the CType
+       Right (EPi i' a body')
+  readBackTyped ctx (VPi aT bT) (VHLam i f) =
+    let x = freshen (ctxNames ctx) (closureName bT) in do
+      b' <- evalClosure bT (VNeutral aT (NVar x))
+      case i of
+           Prim => readBackTyped ctx b' (f VPrimVar) -- TODO double check b' here
```

Comment:
Hmm... shouldn't we get back a lambda from this? Maybe we already do?

Reply:
Good question. Not sure exactly what this reads back to, or what values would cause this case to be hit, but it's pretty similar to Andras's version https://github.com/dhall-lang/dhall-haskell/commit/627a6cdea0170336ff08de34851d8bdf5180571d#diff-fb8fa4dd3e70181d66f30003f0f14a64f9ffb4d382266ddcf15746f09cd6ad91R776